app/models/user.rb

```diff
@@ -1,18 +1,31 @@
 # frozen_string_literal: true
 
 class User < ApplicationRecord
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable, :registerable, and :omniauthable
+  GITHUB_ORGS_QUERY_URL = URI.parse('https://api.github.com/user/orgs')
+
   devise :rememberable, :omniauthable, omniauth_providers: [:github]
 
   validates :login, :uid, presence: true, uniqueness: true
   validates :email, :uid, presence: true, uniqueness: true
 
   def self.from_omniauth(auth)
-    where(provider: auth.provider, uid: auth.uid).first_or_create(
-      email:    auth.info.email,
-      name:     auth.info.name,
-      login:    auth.extra.raw_info.login
+    where(provider: auth.provider, uid: auth.uid).first_or_initialize(
+      email:  auth.info.email,
+      name:   auth.info.name,
+      login:  auth.extra.raw_info.login,
+      token:  auth.credentials.token
     )
   end
+
+  def organisations
+    request = Net::HTTP::Get.new(GITHUB_ORGS_QUERY_URL)
+    request['Authorization'] = "token #{token}"
+    req_options = { use_ssl: GITHUB_ORGS_QUERY_URL.scheme == 'https' }
```

Comment:
I would suggest we just hardcode "always use SSL" requirement.
So this line goes away, and line L#25 becomes more simple (see below)